src/main/java/org/javarosa/core/model/trace/EvaluationTraceReporter.java

```diff
@@ -9,12 +9,28 @@
  *
  * Created by ctsims on 10/19/2016.
  */
-public interface EvaluationTraceReporter {
-    boolean wereTracesReported();
+public abstract class EvaluationTraceReporter {
 
-    void reportTrace(EvaluationTrace trace);
+    public abstract boolean wereTracesReported();
 
-    void reset();
+    public abstract void reportTrace(EvaluationTrace trace);
 
-    Vector<EvaluationTrace> getCollectedTraces();
+    public abstract void reset();
+
+    public abstract Vector<EvaluationTrace> getCollectedTraces();
+
+    public void printAndClearTraces(String description) {
```

Comment:
1. Will needs to overwrite the behavior as the result of the code being run on a different platform, and having to superclass the Reporter (whose behavior differs in how it collects and aggregates the incoming EvaluationTraces) based on the platform would "fork" the reason for the subclass. 

As an example: One of the current trace reporters aggregates inputs and reduces them to time profiles, the other reduces them to debugging traces. If Will wanted both of them to work on FormPlayer, he'd actually have to subclass both of those, and independently have each subclass re-implement this method to use the same behavior.

2. I was referring to this behavior:
https://github.com/dimagi/commcare-core/pull/603/files#diff-ea18a5fcfa3cd726df3085e864d41c2bR113

https://github.com/dimagi/commcare-core/pull/603/files#diff-177f6db6a65942ac830e218ab8fc34f1R111

Which is common and would be re-implemented each time hte profile is used. This call could be put into a static utility method like
https://github.com/dimagi/commcare-core/blob/master/src/main/java/org/javarosa/core/model/utils/InstrumentationUtils.java#L19

but only if the reporter is an argument to the method, it can't be done with polymorphism because the reporter itself is the thing that's null.